common/src/ui/MapDocument.cpp

```diff
@@ -1594,21 +1595,16 @@ void MapDocument::selectFacesWithMaterial(const mdl::Material* material)
 
 void MapDocument::selectBrushesWithMaterial(const mdl::Material* material)
 {
-  const auto brushes = kdl::vec_filter(
-    mdl::collectSelectableNodes(std::vector<mdl::Node*>{m_world.get()}, *m_editorContext),
-    [&](const auto& node) {
-      const auto faces = mdl::collectSelectableBrushFaces(node, *m_editorContext);
-
-      for (const auto faceHandle : faces)
-      {
-        if (faceHandle.face().material() == material)
-        {
-          return true;
-        }
-      }
-
-      return false;
-    });
+  const auto brushes =
+    mdl::collectSelectableNodes(std::vector<mdl::Node*>{m_world.get()}, *m_editorContext)
```

Comment:
Ah, this is one of the cases where you cannot pass an rvalue range to the pipe operator. This should be

```
const auto selectableNodes = mdl::collectSelectableNodes(std::vector<mdl::Node*>{m_world.get()}, *m_editorContext);
const auto brushes = selectableNodes | std::views::filter...
```

Reply:
weird that it compiles on my machine

Reply:
Yes! It's quite annoying and I keep falling into that trap myself. But `std::views::filter` doesn't work with temporaries :-(